src/NewChaiBuffer.hpp

```diff
@@ -49,6 +48,38 @@ inline chai::ArrayManager & getArrayManager()
 // CHAI is not threadsafe so we use a lock to serialize access.
 static std::mutex chaiLock;
 
+inline chai::ExecutionSpace toChaiExecutionSpace( MemorySpace const space )
+{
+  if( space == MemorySpace::NONE )
+    return chai::NONE;
+  if( space == MemorySpace::CPU )
+    return chai::CPU;
+#if defined(USE_CUDA)
+  if( space == MemorySpace::GPU )
+    return chai::GPU;
```

Comment:
Do we even want to use the chai execution spaces? It seems like they should just alias the `MemorySpace`

Reply:
The Chai api needs a `chai::ExecutionSpace` so we need to have a method to convert between the two.